File: frontend/lib/types/project.ts
export interface Project {
  id: string
  name: string
  description: string
  status: 'active' | 'archived'
  task_count: number
  completion_percentage: number
  created_at: string
  updated_at: string
}

export interface CreateProjectInput {
  name: string
  description?: string
}

export interface UpdateProjectInput {
  name?: string
  description?: string
  status?: 'active' | 'archived'
}

export interface ProjectListFilter {
  page?: number
  page_size?: number
  search?: string
}

export interface ProjectStatistics {
  total_tasks: number
  todo_tasks: number
  in_progress_tasks: number
  done_tasks: number
  cancelled_tasks: number
  overdue_tasks: number
  completion_rate: number
}

File: backend/internal/domain/project/project.go
package project

import (
	"time"
)

type Status string

const (
	StatusActive   Status = "active"
	StatusArchived Status = "archived"
)

type Project struct {
	ID                   string    `json:"id"`
	Name                 string    `json:"name"`
	Description          string    `json:"description"`
	Status               Status    `json:"status"`
	TaskCount            int       `json:"task_count"`
	CompletionPercentage float64   `json:"completion_percentage"`
	CreatedAt            time.Time `json:"created_at"`
	UpdatedAt            time.Time `json:"updated_at"`
}

type CreateProjectInput struct {
	Name        string `json:"name" validate:"required,max=100"`
	Description string `json:"description" validate:"max=500"`
}

type UpdateProjectInput struct {
	Name        *string `json:"name" validate:"omitempty,max=100"`
	Description *string `json:"description" validate:"omitempty,max=500"`
	Status      *Status `json:"status" validate:"omitempty,oneof=active archived"`
}

type ProjectListFilter struct {
	Page     int    `form:"page" validate:"min=1"`
	PageSize int    `form:"page_size" validate:"min=1,max=100"`
	Search   string `form:"search" validate:"max=100"`
}

type ProjectStatistics struct {
	TotalTasks      int     `json:"total_tasks"`
	TodoTasks       int     `json:"todo_tasks"`
	InProgressTasks int     `json:"in_progress_tasks"`
	DoneTasks       int     `json:"done_tasks"`
	CancelledTasks  int     `json:"cancelled_tasks"`
	OverdueTasks    int     `json:"overdue_tasks"`
	CompletionRate  float64 `json:"completion_rate"`
}
